pkg/skaffold/build/docker/errors.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/pkg/jsonmessage"
+
+	sErrors "github.com/GoogleContainerTools/skaffold/pkg/skaffold/errors"
+	"github.com/GoogleContainerTools/skaffold/proto"
+)
+
+func newBuildError(err error) error {
+	errU := errors.Unwrap(err)
+	if errU == nil {
+		return err
```

Comment:
What kind of errors are these?  Why do we return them instead of wrapping them with a BUILD_UNKNOWN error?  (is it because they'll be parsed and turned into a error elsewhere?)

Reply:
these are known docker errors. Docker wraps these for known types https://github.com/moby/moby/blob/master/errdefs/defs.go
e.g. here 
https://github.com/moby/moby/blob/master/errdefs/helpers.go#L138